src/baal/calibration/calibration.py

```diff
@@ -0,0 +1,129 @@
+import structlog
+from copy import deepcopy
+import torch
+from torch.utils.data import Dataset
+from torch.optim import Adam
+from torch import nn
+from baal.utils.metrics import ECE
+from baal import ModelWrapper
+
+log = structlog.get_logger("Calibrating...")
+
+
+class DirichletCalibrator(object):
+    """
+    Adding a linear layer to a classifier model after the model is
+    trained and train this new layer until convergence.
+    Together with the linear layer, the model is now calibrated.
+    Source: https://arxiv.org/abs/1910.12656
+
+    Args:
+
+        wrapper (ModelWrapper): Provides training and testing methods.
+        num_classes (int): Number of classes in classification task.
+        lr (float): Learning rate.
+        reg_factor (float): Regularization factor for the linear layer weights.
+        mu (float): Regularization factor for the linear layer biases.
+            If not given, will be initialized by "l".
```

Comment:
"l" == reg_factor?

Reply:
yeah, flake was annoyed by "l" so I changed it to "red_factor"